calico_kubernetes/calico_kubernetes.py

```diff
@@ -181,32 +227,43 @@ def _container_add(self, pid, ip, interface, hostname, orchestrator_id):
         # Obtain information from Docker Client and validate container state
         self._validate_container_state(self.docker_id)
 
-        # Assign ip address through IPAM Client
-        try:
-            ip_assigned = self._datastore_client.assign_address(None, ip)
-        except PoolNotFound:
-            logger.error("IP address %s does not belong to any configured "
-                         "pools. Exiting.", ip)
-            sys.exit(1)
+        if CALICO_IPAM == 'calico':
+            logger.info("Using Calico IPAM")
+            # Assign ip address through IPAM Client
+            try:
+                ip_lists = self._datastore_client.auto_assign_ips(
```

Comment:
instead of assigning the return value to a single variable, you can break it into two vars inline.

```
ipv4_addrs, ipv6_addrs = self._datastore_client.auto_assign_ips( ... )
assert not ipv6_addrs  # We didn't ask for any IPv6 addresses
```